craigslist/__init__.py

```diff
@@ -237,25 +247,53 @@ def customize_result(self, result, html_row):
         """ Add custom/delete/alter fields to result. """
         pass  # Override in subclass to add category-specific fields.
 
-    def geotag_result(self, result):
+    def geotag_result(self, result, soup):
         """ Adds (lat, lng) to result. """
 
         self.logger.debug('Geotagging result ...')
 
-        if result['has_map']:
-            response = requests_get(result['url'], logger=self.logger)
-            self.logger.info('GET %s', response.url)
-            self.logger.info('Response code: %s', response.status_code)
-
-            if response.ok:
-                soup = BeautifulSoup(response.content, 'html.parser')
-                map = soup.find('div', {'id': 'map'})
-                if map:
-                    result['geotag'] = (float(map.attrs['data-latitude']),
-                                        float(map.attrs['data-longitude']))
+        map = soup.find('div', {'id': 'map'})
+        if map:
+            result['geotag'] = (float(map.attrs['data-latitude']),
+                                float(map.attrs['data-longitude']))
 
         return result
 
+    def include_details(self, result, soup):
+        """ Adds description, images to result """
+
+        self.logger.debug('Adding details to result...')
+
+        body = soup.find('section', {'id': 'postingbody'})
+        result['body'] = body.text.strip()
+
+        img_matcher = re.compile(
```

Comment:
How about something like this? It preserves the order that the images are presented on the detail page:

```python
image_tags = soup.find_all('img')
images = [None] * len(image_tags)
for img in image_tags:
    img_link = img['src'].replace('50x50c', '600x450')
    index = int(img['alt'])
    images[index - 1] = img_link

result['images'] = filter(None, images)
```

The challenge is that one image will always be given twice, as a thumbnail and full size. Maybe we can assume that it will always be the first image that has this, and just skip it and create a list from the others?

Here's an example output with just the image tags:

```html
<img src="https://images.craigslist.org/01313_asaTEP4kqvn_600x450.jpg" title=" 1"
 alt=" 1">
<img src="https://images.craigslist
.org/01313_asaTEP4kqvn_50x50c.jpg" class="selected" alt=" 1">
<img src="https://images.craigslist.org/00707_7APlhhtZABl_50x50c.jpg"  alt=" 2">
<img src="https://images.craigslist.org/00505_96Yo7YiCIvv_50x50
c.jpg"  alt=" 3">
```